File: shopapp-angular-prefix/shopapp-angular/src/app/dtos/user/changes.password.dto.ts
export class ChangesPasswordDto{
  old_password: string;
  password: string;
  retype_password: string;

  constructor(data: any) {
    this.password = data.password;
    this.old_password = data.old_password;
    this.retype_password = data.retype_password;
  }
}

File: shopapp-backend/shopapp/src/main/java/com/project/shopapp/dtos/ChangesPasswordDTO.java
package com.project.shopapp.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import lombok.*;
import org.springframework.stereotype.Component;

import java.util.Date;

@Getter
@Setter
@Data
@Builder
@Component
@AllArgsConstructor
@NoArgsConstructor
public class ChangesPasswordDTO {
    @NotBlank
    @JsonProperty("old_password")
    private String oldPassword;

    @JsonProperty("password")
    private String password;

    @JsonProperty("retype_password")
    private String retypePassword;

}
